yarn-project/aztec.js/src/ethereum/portal_manager.ts

```diff
@@ -437,12 +437,14 @@ export class L1TokenPortalManager extends L1ToL2TokenPortalManager {
    * @param l2Bridge - Address of the L2 bridge.
    * @param callerOnL1 - Caller address on L1.
    */
-  public getL2ToL1MessageLeaf(
+  public async getL2ToL1MessageLeaf(
     amount: bigint,
     recipient: EthAddress,
     l2Bridge: AztecAddress,
     callerOnL1: EthAddress = EthAddress.ZERO,
-  ): Fr {
+  ): Promise<Fr> {
+    const version = await this.outbox.read.VERSION();
+
```

Comment:
If we need the rollup contract, lets make in the constructor - getVersion will be memoized so it will only need to fetch it once, if we construct it each time we will need to request it each time

Reply:
Rollup contract was removed from there (it increased bundle size) and the outbox got the data directly. Theres no wrapper of the outbox so just did it this way

Reply:
ok